### DISCUSSION TITLE
How about transition mobile route to a more official package?

### DISCUSSION THREAD

**Discussion by xpwmaosldk** [immich-app/immich#275] [General]
Hi there!
I appreciate that I have the talent to contribute.
And I am delighted to be joining this great project with you.

By the way let's get to the point.
I think it need to transition mobile route to a more official package.
`auto_route` is works perfectly but it is community driven.
Although it is a small possibility, it could cause the project to freeze or be scrapped.
I thought it would be good to change it as early as possible
So I suggest a bit more official package `go_router`([here](https://pub.dev/packages/go_router))
This is managed by the flutter team and does not require build_runner.

If approved, I will proceed with this operation.

Tell me your opinions.

Thanks!

**Comment by alextran1502** [comment]
If it is not too much work, you are welcomed to open a PR and show the implementation of this package. I am interested in seeing how does go_router handle tab page

**Reply by xpwmaosldk** [reply]
I need some time because I'm working with my job. 💻

**Reply by alextran1502** [reply]
No rush here , your job is more important 😄

**Comment by xpwmaosldk** [comment]
https://user-images.githubusercontent.com/13146337/176867538-4a01888b-bb12-4dfd-aaa3-09d64130b09b.mov

Very tiny demo of go_router.

There are still many side effects left. I need a few weeks.

FYI. I changed bottom icon.(ref: https://pub.dev/packages/font_awesome_flutter)
how is it?😄

**Reply by alextran1502** [reply]
I noticed that each page reload when you tap on each tab. It is not the current behavior of the app. So if you scroll down at a certain point of the timeline, then you navigate to the search page and back, it should stay at that location of the timeline. 

About the new icon, look good but I try to have the UI identical to Google Photos and avoid adding additional dependencies when not needed 😜

**Reply by xpwmaosldk** [reply]
I strongly agree that it doesn't add unnecessary dependencies.
I'll put this back.

And thank you for your point.
This is a very small implementation.
Eventually I plan to implement saving the page state ☺️
